### PR TITLE
a script to retrieve harmony cpu usage info

### PR DESCRIPTION
**Pull request by AndyBoWu**
bash file needed to be running in the nodes

### CODE REVIEW

**Comment by AndyBoWu** [comment]
Okay, Let me re-deploy on a Linux machine later, and share the test logs with you.

**Comment by AndyBoWu** [comment]
URL_harmony_top=https://raw.githubusercontent.com/AndyBoWu/experiment-deploy-1/master/configs/better-top.sh
curl -LO $URL_harmony_top
chmod +x better-top.sh

**Comment by AndyBoWu** [comment]
Looks fine on my side. I can curl the file from the above url, then chmod the file, and it can run without any issue on a test Amazon Linux machine.

File: configs/better-top.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+z=$(ps -aux | grep harmony)
+while read -r z
+do
+   var=$var$(awk '{print "cpu_usage{process=\""$11"\", pid=\""$2"\"}", $3z}');
+done <<< "$z"
+curl -X POST -H  "Content-Type: text/plain" --data "$var
```
Comment (LeoHChen) [inline configs/better-top.sh]: really? two lines?  why don't you put the $var into a file?

File: configs/better-top.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+z=$(ps -aux | grep harmony)
```
Comment (LeoHChen) [inline configs/better-top.sh]: you have to filter out grep command itself, otherwise, you will always have two processes, one harmony, and one grep.

**Comment by LeoHChen** [comment]
don't put the better-top into experiment-deploy repo

**Comment by AndyBoWu** [comment]
The better-top.sh file is located here: `harmony-one/harmony-ops/pushgateway/`

PR created.

**Comment by AndyBoWu** [comment]
Added `curl` ERROR exception handling for downloading node_exporter, pushgateway, and better_top.sh

File: configs/better-top.sh (hunk 2 above)
Comment (AndyBoWu) [reply configs/better-top.sh]: It does return two entries. 

The cmd does a better job! 

$ps aux | egrep '[h]armony'

**Comment by AndyBoWu** [comment]
Hi @LeoHChen , systemd_harmony is the function of interest to make harmony running as a service.

File: tf/gcp/main.tf [hunk 3]
```diff
@@ -0,0 +1,96 @@
+// Configure the Google Cloud provider
+// https://cloud.google.com/community/tutorials/getting-started-on-gcp-with-terraform
+provider "google" {
+  credentials = "${file("flowing-digit-248617-6d1dac5cb46e.json")}"
+  project     = "flowing-digit-248617"
+  region      = "us-west1"
```
Comment (LeoHChen) [inline tf/gcp/main.tf]: make it a variable.

File: tf/gcp/main.tf [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  byte_length = 8
+}
+
+resource "google_compute_firewall" "default" {
+   name    = "flask-app-firewall"
+   network = "default"
+
+   allow {
+     protocol = "tcp"
+     ports    = ["5000"]
```
Comment (LeoHChen) [inline tf/gcp/main.tf]: only 5000? why? other ports?

File: tf/gcp/main.tf [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+   }
+
+}
+
+// A single Google Cloud Engine instance
+resource "google_compute_instance" "default" {
+ name         = "flask-vm-${random_id.instance_id.hex}"
+ machine_type = "f1-micro"
```
Comment (LeoHChen) [inline tf/gcp/main.tf]: what's the spec of the f1-micro machine_type?

File: tf/gcp/main.tf [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+ zone         = "us-west1-a"
```
Comment (LeoHChen) [inline tf/gcp/main.tf]: make it a variable.

File: tf/gcp/main.tf [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+ boot_disk {
+   initialize_params {
+     image = "debian-cloud/debian-9"
+   }
+ }
+
+
+// Make sure flask is installed on all new instances for later steps
+// https://stackoverflow.com/questions/36591653/how-do-i-use-the-file-provisioner-with-google-compute-instance-template
+ metadata_startup_script = "sudo apt-get update; "
```
Comment (LeoHChen) [inline tf/gcp/main.tf]: do you need to install bind-utils for the dig command?

File: tf/gcp/main.tf [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        connection {
+            host = "${google_compute_instance.default.network_interface.0.access_config.0.nat_ip}"
+            type = "ssh"
+            user = "gce-user"
+            private_key = "${file("~/.ssh/id_rsa")}"
+        }
+    }
+
+    provisioner "file" {
+        source = "files/harmony.service"
```
Comment (LeoHChen) [inline tf/gcp/main.tf]: you need to check in the harmony.service as well.

File: tf/gcp/main.tf [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+            host = "${google_compute_instance.default.network_interface.0.access_config.0.nat_ip}"
+            type = "ssh"
+            user = "gce-user"
+            private_key = "${file("~/.ssh/id_rsa")}"
+        }
+
+    } 
+
+metadata = {
+   ssh-keys = "gce-user:${file("~/.ssh/id_rsa.pub")}"
```
Comment (LeoHChen) [inline tf/gcp/main.tf]: we should use the same key pair for all nodes.  Try to use the same file as my aws script.

**Pull request closed without merging**